painting events (Spyder)

JaskRendix opened the pull request:
PR adds painting events.

money_is part of #1726 

Painting events between Cotton and Candy are done. The player can buy: starry or monsters or trepidation or bundle (all 3).

the player can buy the bundle (all three) or each one by clicking on the painting. When bought it'll replace the text with.
![Screenshot from 2023-04-18 07-58-48](https://user-images.githubusercontent.com/64643719/232685040-52e81e88-d3e6-4b40-9c5e-78eacd42cffb.png)
added this here
![Screenshot from 2023-04-18 07-57-39](https://user-images.githubusercontent.com/64643719/232685046-1ce3d6a5-0b45-40f9-b5ad-2d5b5b711e37.png)
if the player buys all three paintings
![Screenshot from 2023-04-18 08-03-59](https://user-images.githubusercontent.com/64643719/232685824-3b1bc347-e458-4402-a54c-49e208440e85.png)

JaskRendix commented:
@Sanglorian tell me if it's ok

proposals:
- do you like the idea of adding a meeting (no fight) with Billie while she is going out of the Art Shop? (the dialogue could be around the painting she bought / new capture / something else).
- do you agree that if the player buys a painting he/she doesn't have to pay anymore the fee (50)?

Sanglorian commented:
> @Sanglorian tell me if it's ok
> 
> proposals:
> 
>     * do you like the idea of adding a meeting (no fight) with Billie while she is going out of the Art Shop? (the dialogue could be around the painting she bought / new capture / something else).
> 
>     * do you agree that if the player buys a painting he/she doesn't have to pay anymore the fee (50)?

Yeah, those both sound good!

Longer term it could be good to have the paintings disappear when bought (although I like the idea of Billie having one, maybe it could have been bought by her but donated back to the gallery? Then you could annoy the player by having an NPC talk about how generous Billie is). 

Obviously removing the paintings from display would require #666 to be implemented, so I think your solution of saying it's a copy is a great one in the meantime.

JaskRendix commented:
> Obviously removing the paintings from display would require https://github.com/Tuxemon/Tuxemon/issues/666 to be implemented, so I think your solution of saying it's a copy is a great one in the meantime.

Yeah, I'm aware of this, and I was thinking about too, but the solution doesn't go far than your NPC workaround.
Probably with the only difference that there will be an action called **create_item** (like create_npc)

> I like the idea of Billie having one, maybe it could have been bought by her but donated back to the gallery

That's good

> Yeah, those both sound good!

Since you're Billie's voice, then I need some dialogues!

JaskRendix commented:
@Sanglorian 

> Obviously removing the paintings from display would require https://github.com/Tuxemon/Tuxemon/issues/666 to be implemented, so I think your solution of saying it's a copy is a great one in the meantime.

Yeah, I'm aware of this, and I was thinking about too, but the solution doesn't go far than your NPC workaround.
Probably with the only difference that there will be an action called **create_item** (like create_npc)

> I like the idea of Billie having one, maybe it could have been bought by her but donated back to the gallery

That's good.
- added the donated back to the gallery
- added the exclusion to pay if buy a painting.

> Yeah, those both sound good!

Since you're Billie's voice, then I need some dialogues!

Sanglorian commented:
> Since you're Billie's voice, then I need some dialogues!

Oh, yes! How about:

"I see you're trying to broaden your horizons. That's cute. Just try not to touch anything."

JaskRendix commented:
That's on point! I like it!

JaskRendix committed:
painting

JaskRendix commented:
Everything is done. Ready to go.
